File: Programmers/60057.py
def compression(s, num):
    result = ''
    prev = ''
    count = 1
    for i in range(0, len(s), num):
        curr = s[i:i+num]
        if prev == curr:
            count += 1
        else:
            if count == 1:
                result += prev
            else:
                result += '%d%s'%(count, prev)
            count = 1
            prev = curr
    if count == 1:
        result += prev
    else:
        result += '%d%s'%(count, prev)

    return result


def solution(s):
    if len(s) == 1:
        return 1
    answer = float('inf')
    for num in range(1, len(s) // 2 + 1):
        answer = min(answer, len(compression(s, num)))
    return answer

File: Programmers/60057.js
function compression(s, num) {
    const splitted = [];
    for(let i = 0; i < s.length; i+= num) {
        splitted.push(s.substring(i, i+num));
    }
    let compressed = '';
    let count = 1;
    for(let i = 0; i < splitted.length - 1; i++) {
        if (splitted[i] === splitted[i+1]) {
            count += 1;
        }
        else {
            if(count > 1){
                compressed += count;
                count = 1;
            }
            compressed += splitted[i];
        }
    }
    if(count > 1) {
        compressed += count;
    }
    compressed += splitted[splitted.length-1];
    return compressed.length;
}

function solution(s) {
    let answer = Infinity;
    for(let num = 1; num < s.length/2 + 1; num++){
        answer = Math.min(answer, compression(s, num));
    }
    return answer;
}